src/app.js

```diff
@@ -7,6 +7,19 @@ export function app(app) {
   var node
   var element
 
+  function hydrate(elm) {
+    return elm == null ? undefined : {
+      tag: elm.tagName,
+      data: {},
+      children: [].map.call(elm.childNodes, function(child){hydrate(child)})
+    }
+  }
+
+  if (root.hasChildNodes !== undefined && root.hasChildNodes()) {
+    element = root.children[0]
+    node = hydrate(element)
+  }
+
```

Comment:
I know it won't be the most elegant, but instead of requiring some dom attribute to differentiate, I think adding a flag for hydrate to the call to app is cleaner.

```js
app({state, views, mixins, hydrate: true})
```
This way by default the current behavior can be applied, and then if the user is using SSR they can opt-in to hydrate the dom.  However, I am open to whatever approach you all feel is best.  I just really want to be able to avoid duplicating dom or flickering on first client side renders for my use case.  :smile: